packages/toolpad-app/src/toolpad/AppEditor/StructureExplorer/index.tsx

```diff
@@ -187,40 +213,74 @@ export default function PageStructureExplorer() {
 
   const handleNodeToggle = React.useCallback(
     (event: React.SyntheticEvent, nodeIds: string[]) => {
-      setExpandedDomNodeIds((prev) => {
-        const newIds = nodeIds.filter((id) => !prev.includes(id));
-        const retainedIds = prev.filter((id) => nodeIds.includes(id));
-        return [...retainedIds, ...newIds];
-      });
+      setExpandedDomNodeIds(nodeIds);
     },
     [setExpandedDomNodeIds],
   );
 
+  const handleNodeDelete = React.useCallback(
+    (event: React.KeyboardEvent<HTMLUListElement>) => {
+      // delete selected node if event.key is Backspace
+      if (event.key === 'Backspace') {
+        appStateApi.update(
+          (draft) => {
+            if (!selectedDomNodeId) {
+              return draft;
+            }
+            const toRemove = appDom.getMaybeNode(dom, selectedDomNodeId);
+            if (toRemove && appDom.isElement(toRemove)) {
+              draft = appDom.removeNode(draft, toRemove.id);
+              draft = deleteOrphanedLayoutNodes(dom, draft, toRemove);
+            }
+            return draft;
+            // return normalizePageRowColumnSizes(draft);
+          },
+          currentView.kind === 'page'
```

Comment:
I wonder if we can get rid of these `currentView.kind === 'page'` as I think these actions can only occur in a page view? But I may have forgotten why I added these checks in the other similar deletes...

Reply:
I think it's for the types, in which case I can probably replace these conditionals with 
```...(currentView as Extract<DomView, { kind: 'page' }>),```